etk/extractors/dbpedia_spotlight_extractor.py

```diff
@@ -1,25 +1,66 @@
-from etk.extractor import Extractor
+from etk.extractor import Extractor, InputType
 from etk.extraction import Extraction
 from typing import List
-
+import requests
 
 
 class DBpediaSpotlightExtractor(Extractor):
-    def __init__(self):
+    search_url = ''
+
+    def __init__(self, extractor_name: str, url: str):
+        Extractor.__init__(self, input_type=InputType.TEXT,
+                           category="build_in_extractor",
+                           name=extractor_name)
+        self.search_url = url
         # TODO initialise stuff
         # TODO if using the dbpedia webservice, a class variable should be set with the url
         # TODO if using the codebase, the library can be setup to called here
         # TODO if using the codebase, we need a local dbpedia against which the queries are to be made, set it up here as well
-        pass
 
-    def extract(self, text: str) -> List[Extraction]:
+    def extract(self, text: str, confidence: float, filter: list) -> List[Extraction]:
```

Comment:
confidence and filter should be optional parameters with default value of 0.5 and [Person, Organisation, Place] respectively. `extract` methods in all extractors have the same signature, with optional parameters